Sources/MacroToolkit/Type.swift

```diff
@@ -174,6 +169,199 @@ public enum `Type`: TypeProtocol, SyntaxExpressibleByStringInterpolation {
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            
+            let normalizedConstraint = Type(implicitlyUnwrappedOptionalTypeSyntax.wrappedType).normalized()
+            implicitlyUnwrappedOptionalTypeSyntax.wrappedType = TypeSyntax(normalizedConstraint._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(implicitlyUnwrappedOptionalTypeSyntax)
+            
+            return .implicitlyUnwrappedOptional(.init(implicitlyUnwrappedOptionalTypeSyntax, attributedSyntax: attributedTypeSyntax))
+            
+        case .member(let type):
+            var memberTypeSyntax: MemberTypeSyntax = type._baseSyntax
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            let normalizedBaseType = Type(type._baseSyntax.baseType).normalized()
+            
+            memberTypeSyntax.genericArgumentClause = memberTypeSyntax.genericArgumentClause?.normalized()
+            
+            memberTypeSyntax.baseType = TypeSyntax(normalizedBaseType._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(memberTypeSyntax)
+            
+            return .member(.init(memberTypeSyntax, attributedSyntax: attributedTypeSyntax))
+            
+        case .metatype(let type):
+            let baseType = type._baseSyntax
+            let memberTypeSyntax = MemberTypeSyntax.init(
+                leadingTrivia: baseType.leadingTrivia,
+                baseType: baseType.baseType,
+                name: baseType.metatypeSpecifier,
+                trailingTrivia: baseType.trailingTrivia
+            )
+            if var attributedSyntax = type._attributedSyntax {
+                attributedSyntax.baseType = TypeSyntax(memberTypeSyntax)
+                return .member(.init(memberTypeSyntax, attributedSyntax: attributedSyntax))
+            } else {
+                return .member(.init(memberTypeSyntax))
+            }
+        case .missing(let type):
+            return .missing(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .optional(let type):
+            var optionalTypeSyntax: OptionalTypeSyntax = type._baseSyntax
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            let normalizedElement = Type(optionalTypeSyntax.wrappedType).normalized()
+            optionalTypeSyntax.wrappedType = TypeSyntax(normalizedElement._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(optionalTypeSyntax)
+            
+//            let identifierSyntax = IdentifierTypeSyntax(
+//                leadingTrivia: optionalTypeSyntax.leadingTrivia,
+//                name: .identifier("Optional"),
+//                genericArgumentClause: .init(
+//                    arguments: .init(arrayLiteral: .init(argument: optionalTypeSyntax.wrappedType))
+//                ),
+//                trailingTrivia: optionalTypeSyntax.leadingTrivia
+//            )
```

Comment:
Why doesn't this approach work? It'd be preferable but I assume there was something about it that caused you to try the approach below instead.

Reply:
This approach does actually work, the other one was just less code and most likely was as effective 99% of the time. I now switched to this more verbose approach that maintains trivia information with 206fe0a.